### ISSUE TITLE
[FEATURE] Clear response option for survey questions that are optional

### ISSUE THREAD

**Issue by aryamantodkar** [formbricks/formbricks#4246]
### Is your feature request related to a problem? Please describe.

When having a survey with questions that are not mandatory, user should be able to clear their response and should be able to submit an empty response to that respective question.

![Screenshot from 2024-11-07 06-06-18](https://github.com/user-attachments/assets/3f7cea17-037f-41f4-9fb2-a02e6e4cdd71)

### Describe the solution you'd like

Add a Clear button to clear the response for the respective optional question.

![Screenshot from 2024-11-07 06-28-57](https://github.com/user-attachments/assets/9c7d5298-e2c4-4949-89fd-542853ee68ec)

### Additional context

If you have any suggestions or feedback on a suitable style and placement, please share! Additionally, if there are any existing styles in the project that we should match, please let me know.

**Comment by khAja13** [comment]
@jobenjada Let me know whether I can work on this?

**Comment by khAja13** [comment]
@aryamantodkar Maybe instead of putting a clear button, if a user clicks once more on the option, then we can clear that.

**Comment by aryamantodkar** [comment]
@khAja13 Yes, clicking on the response again to deselect the option will work as well. However I believe having a dedicated 'Clear Response' button might be more intuitive for users.

**Comment by aryamantodkar** [comment]
@khAja13 I plan to work on this but I am waiting for the approval to go ahead.
